lib/foreman_inventory_upload/async/upload_report_job.rb

```diff
@@ -49,6 +52,31 @@ def rh_credentials
           }
         end
       end
+
+      def cdn_proxy
+        cdn_settings = SETTINGS[:katello][:cdn_proxy] || {}
+
+        return nil unless cdn_settings[:host]
+
+        proxy_uri = URI('')
+
+        original_uri = URI.parse(cdn_settings[:host])
+
+        proxy_uri.scheme = original_uri.scheme || 'http'
```

Comment:
@ShimShtein I know this has been merged already, but according to https://github.com/Katello/katello/blob/master/app/lib/katello/util/proxy_uri.rb scheme might be "proxy" or "proxys", which I believe curl won't understand.

See also proxy-related code in older insights plugin: https://github.com/redhataccess/foreman-plugin/commit/48b7583dc7f0372f254e71da067fa146b9c7bb19

Reply:
According to [Katello](https://github.com/Katello/katello/blob/675d54134c23e358b500ed5d9f6e603fc033b5ab/app/lib/katello/util/http_proxy.rb) `proxy` and `proxys` is used only internally. I don't think users know about those schemes at all. Since I don't work with HttpClient here, the issue should not affect me here.